src/bin/mir-json-rustc-wrapper.rs

```diff
@@ -49,28 +49,30 @@ impl rustc_driver::Callbacks for GetOutputPathCallbacks {
 
     fn after_parsing<'tcx>(
         &mut self,
-        _compiler: &Compiler,
+        compiler: &Compiler,
         queries: &'tcx Queries<'tcx>,
     ) -> Compilation {
-        // This phase runs with `--cfg crux`, so some `#[crux_test]` attrs may be visible.  Even
+        // This phase runs with `--cfg crux`, so some `#[crux::test]` attrs may be visible.  Even
         // the limited amount of compilation we do will fail without the injected `register_attr`.
         analyz::inject_attrs(queries);
-        Compilation::Continue
-    }
 
-    fn after_analysis<'tcx>(
-        &mut self,
-        compiler: &Compiler,
-        queries: &'tcx Queries<'tcx>,
-    ) -> Compilation {
         let sess = compiler.session();
-        let crate_name = queries.crate_name().unwrap().peek();
-        let outputs = queries.prepare_outputs().unwrap().peek();
+        let (crate_name, outputs) = {
+            // rustc_session::find_crate_name - get the crate with queries.expansion?
+            let krate = queries.parse().unwrap();
+            let krate = krate.borrow();
+            let crate_name = rustc_session::output::find_crate_name(&sess, &krate.attrs);
+            // get global ctxt then call methods inside of enter() to get
+            let outputs = compiler.build_output_filenames(&sess, &krate.attrs);  //queries.prepare_outputs().unwrap().peek();
+            (crate_name, outputs)
+        };
```

Comment:
```suggestion
        let (crate_name, outputs) = {
            // rustc_session::find_crate_name - get the crate with queries.expansion?
            let krate = queries.parse().unwrap();
            let krate = krate.borrow();
            let crate_name = rustc_session::output::find_crate_name(&sess, &krate.attrs);
            let outputs = compiler.build_output_filenames(&sess, &krate.attrs);
            (crate_name, outputs)
        };
```
`global_ctxt` is not used here, so I think the "get global ctxt ..." comment is outdated.